test/BuildingRegistry.Console.ReactivatedHouseNumbers/Program.cs

```diff
@@ -0,0 +1,253 @@
+                            {
+                                if (hasEndDate && !terrainObjectHouseNumber.EndDateTime.HasValue) //was retired, now not anymore
+                                {
+                                    var subaddresses = GetSubaddressesByStreamAndTerrainObjectHouseNumber(sqlConnection, streamId, terrainObjectHouseNumber, eventsJsonSerializerSettings);
+
+                                    var deletedSubaddresses = subaddresses
+                                        .Where(subaddress => subaddress.Modification == CrabModification.Delete)
+                                        .Select(x => x.SubaddressId)
+                                        .Distinct()
+                                        .ToList();
+
+                                    var nonDeletedSubaddresses = subaddresses
+                                        .Where(subaddress =>
+                                                    subaddress.Timestamp < terrainObjectHouseNumber.Timestamp &&
+                                                    subaddress.Timestamp > timestampSinceLastReactivation)
+                                        .Select(subaddress => subaddress.SubaddressId)
+                                        .Distinct()
+                                        .Except(deletedSubaddresses)
+                                        .ToList();
+
+                                    CreateCommandsIfNecessary(nonDeletedSubaddresses, subaddresses, commandsJsonSerializerSettings);
+
+                                    hasEndDate = false;
+                                    timestampSinceLastReactivation = terrainObjectHouseNumber.Timestamp;
+                                }
+
+                                if (terrainObjectHouseNumber.EndDateTime.HasValue) //need timestamp first occurred
+                                    hasEndDate = true;
+                            }
+                        }
+                    }
+                });
+
+                lastStreamId = streamsIds.Last();
+                streamsIds = GetNextStreamsIds(eventsConnectionString, lastStreamId).ToList();
+
+            } while (streamsIds.Any());
+
+            ReadFilesAndSendCommands(commandsJsonSerializerSettings, appSettings);
+        }
+
+        private static void ReadFilesAndSendCommands(
+            JsonSerializerSettings commandsJsonSerializerSettings,
+            ApplicationSettings appSettings)
+        {
+            int count = 0;
+            var commandsToSend = new List<RegisterCrabImportRequest[]>();
+
+            foreach (var file in Directory.GetFiles(FilesToProcessPath))
+            {
+                count++;
+
+                var command =
+                    JsonConvert.DeserializeObject<RegisterCrabImportRequest>(File.ReadAllText(file), commandsJsonSerializerSettings);
+                commandsToSend.Add(new[] {command});
```

Comment:
Would it be a good idea to move the file to another folder after you have read it in, to make sure if it crashes, you dont have to resend all? And only check the last 20 in your other folder manually.

Reply:
Good point, I removed the batching and instead do a send command each file. And moved the file after it's sent.